dataprofiler/profilers/profile_builder.py

```diff
@@ -1420,26 +1424,32 @@ def _merge_correlation(self, other):
         if corr_mat1 is None or corr_mat2 is None:
             return None
 
-        if len(corr_mat1.columns) != len(corr_mat2.columns) or \
-                len(corr_mat1.columns) <= 1:
+        col_ids1 = np.where(~np.isnan(corr_mat1).all(axis=0))[0]
+        col_ids2 = np.where(~np.isnan(corr_mat2).all(axis=0))[0]
+
+        if len(col_ids1) != len(col_ids2) or len(col_ids1) <= 1:
             return None
-        if set(corr_mat1.columns) != set(corr_mat2.columns):
+        if (col_ids1 != col_ids2).any():
             return None
 
         mean1 = np.array(
             [self._profile[profile_name].profile['statistics']['mean']
-             for profile_name in corr_mat1.columns])
+             for i, profile_name in enumerate(self._profile.keys())
+             if i in col_ids1])
         std1 = np.array(
             [self._profile[profile_name].profile['statistics']['stddev']
-             for profile_name in corr_mat1.columns])
+             for i, profile_name in enumerate(self._profile.keys())
+             if i in col_ids1])
         n1 = self.total_samples
 
         mean2 = np.array(
             [other._profile[profile_name].profile['statistics']['mean']
-             for profile_name in corr_mat1.columns])
+             for i, profile_name in enumerate(self._profile.keys())
+             if i in col_ids1])
         std2 = np.array(
             [other._profile[profile_name].profile['statistics']['stddev']
-             for profile_name in corr_mat1.columns])
+             for i, profile_name in enumerate(self._profile.keys())
+             if i in col_ids1])
```

Comment:
Should this be col_ids2?

Reply:
at this part of code, col_ids1 should be the same as col_ids2, but I changed it for readability

Reply:
changed